management/server/idp/util.go

```
@@ -69,3 +71,24 @@ func baseURL(rawURL string) string {
 
 	return parsedURL.Scheme + "://" + parsedURL.Host
 }
+
+// Provides the env variable name for use with idpTimeout function
+const (
+	idpTimeoutEnv = "NETBIRD_IDP_TIMEOUT"
+)
+
+// idpTimmeout returns a timeout value for the IDP
+func idpTimeout() time.Duration {
+	timeoutStr, ok := os.LookupEnv(idpTimeoutEnv)
+	if !ok || timeoutStr == "" {
+		defaultTimeout, _ := time.ParseDuration("10s")
+		return defaultTimeout
+	}
+
+	timeout, err := time.ParseDuration(timeoutStr)
+	if err != nil {
+		defaultTimeout, _ := time.ParseDuration("10s")
+		return defaultTimeout
+	}
```

You can define a `defaultTimeout` constant once at the top and avoid re-parsing  every time. This also removes the unnecessary ignored error from `time.ParseDuration`.

```go
 defaultTimeout  = 10 * time.Second
```

---

Done